src/analysis/convex/independent.lean

```diff
@@ -0,0 +1,226 @@
+/-
+Copyright (c) 2021 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import analysis.convex.caratheodory
+import analysis.convex.extreme
+import linear_algebra.affine_space.independent
+
+/-!
+# Convex independence
+
+This file defines convex independent families of points.
+
+Convex independence is closely related to affine independence. In both cases, no point can be
+written as a combination of others. When the combination is affine (that is, any coefficients), this
+yields affine independence. When the combination is convex (that is, all coefficients are
+nonnegative), then this yields convex independence. In particular, affine independence implies
+convex independence.
+
+## Main declarations
+
+* `convex_independent p`: Convex independence of the indexed family `p : ι → E`. Every point of the
+  family only belongs to convex hulls of sets of the familty containing it.
+* `convex_independent_iff_finset`: Carathéodory's theorem allows us to only check finsets to
+  conclude convex independence.
+* `convex.extreme_points_convex_independent`: Extreme points of a convex set are convex independent.
+
+## References
+
+* https://en.wikipedia.org/wiki/Convex_position
+
+## TODO
+
+Once convexity is generalised to vector spaces, `convex_independent` will take as first parameter
+the semiring of scalars, so that `convex_independent p` becomes `convex_independent 𝕜 p`, mimicking
+`affine_independent 𝕜 p`. Our current definition would then correspond to `convex_independent ℝ p`.
+
+Prove `affine_independent.convex_independent`. This requires some glue between `affine_combination`
+and `finset.center_mass`.
+
+## Tags
+
+independence, convex position
+-/
+
+open_locale affine big_operators classical
+open finset function
+
+--variables (𝕜 : Type*) {E : Type*} [ordered_semiring 𝕜] [add_comm_group E] [module 𝕜 E]
```

Comment:
I think if you want to leave comments that are useful for a future generalisation, you should explictly flag each on with a text comment.

Reply:
Sure, but I'm planning on carrying out the generalisation in the following week anyway.